tests/pgautofailover_utils.py

```diff
@@ -225,10 +232,19 @@ def connection_string(self):
                  self.port,
                  self.database)
 
+        # If a local CA is used, or even a self-signed certificate,
+        # using verify-ca often provides enough protection.
         if self.sslMode:
-            # If a local CA is used, or even a self-signed certificate,
-            # using verify-ca often provides enough protection.
-            dsn += "?sslmode=%s" % self.sslMode
+            sslMode = self.sslMode
+        elif self.sslSelfSigned:
+            sslMode = "require"
+        else:
+            sslMode = "prefer"
+
+        dsn += "?sslmode=%s" % sslMode
+
+        if self.sslCAFile:
+            dsn += f"&sslrootcert={self.sslCAFile}"
```

Comment:
Do we need to add the client root certificate and revocation list when those exist?